lib/utils.c

```diff
@@ -39,6 +39,56 @@
 
 int timestamp_short;
 
+int read_prop(char *dev, char *prop, long *value)
+{
+	char fname[128], buf[80], *endp;
+	int status;
+	FILE *fp;
+	long result;
+	int ret;
+
+	ret = snprintf(fname, sizeof(fname), "/sys/class/net/%s/%s", dev, prop);
+	if (ret <= 0 || ret >= sizeof(fname))
+		goto out;
```

Comment:
I realised that `snprintf()` could very rarely go wrong here, and in that case: similar code in the repository where its used, `snprintf()` doesn't check for errors, and in the rare cases that it does, it just returns -1 (because we really can't be sure what went wrong in this case). Do let me know how we proceed, thanks.